### ISSUE TITLE
Explicitly set entry point for Deno, to not load browser version by default

### ISSUE THREAD

**Issue by Stephan-C** [rollup/rollup#4398]
### Feature Use Case

The assumption made here: https://github.com/rollup/rollup/pull/3634 was to load the browser version by default for Deno targets. However, this is not always the best scenario. There are many cases where the browser version will need access to the DOM, where the node (server) version or an alternative entry point will be more suited for loading into Deno.

### Feature Proposal

Add a way to explicitly set the the entry point for Deno targets, which overrides loading the browser entry point on Deno.

**Comment by Stephan-C** [comment]
I didn't say rollup needs access to the DOM. I said the browser entry point setting is the default entry point on Deno when loading a library that was packaged with Rollup. There should be some override to specify the entry point for a Deno target, explicitly.

You can load npm packages into Deno using, e.g. Skypack, jspm.io, jsDelivr, esm.sh, etc.

**Comment by lukastaegert** [comment]
> the browser entry point setting is the default entry point on Deno when loading a library that was packaged with Rollup

How does Rollup do that when it only writes your JavaScript chunks and does not write e.g. your package.json file? Are you sure you are not confusing Rollup core with some Rollup plugin?

**Comment by lukastaegert** [comment]
> I said the browser entry point setting is the default entry point on Deno when loading a library that was packaged with Rollup. There should be some override to specify the entry point for a Deno target, explicitly.

How would Rollup set the entry point of a package? Rollup core is only outputting JavaScript by default, it does not write e.g. package.json files. I have not the slightest idea what you actually expect Rollup to do.

**Comment by Stephan-C** [comment]
Yeah, sorry, this might be something in a rollup plugin, I will have a look at the plugins used. The only reason why I linked that PR, is because that is the only place I could find a reference to the browser version loading by default in Deno, https://github.com/rollup/rollup/pull/3634#issuecomment-643811723 . Nowhere else I could find something similar to the issue I'm trying to solve. I'm still new to Rollup and trying to find solutions for this issue: https://github.com/remarkablemark/html-dom-parser/issues/181

**Comment by guybedford** [comment]
If you add a `"deno"` condition in the exports field, and configure this with rollup-plugin-node-resolve as an exports condition, that is one technique that can work here and that we use in JSPM.

**Comment by frank-dspeed** [comment]
@lukastaegert good news i have one time refactored rollup for use with deno it was relativ easy i splited the node dependencies into a own package and i import them from there after that i created the same package with the same api's but used the deno compatible dependencies it is not much needed FS Path thats it deno can do the rest out of the box
